src/analysis/normed_space/normed_group_hom.lean

```diff
@@ -458,6 +461,26 @@ def range : add_subgroup V₂ := f.to_add_monoid_hom.range
 lemma mem_range (v : V₂) : v ∈ f.range ↔ ∃ w, f w = v :=
 by { rw [range, add_monoid_hom.mem_range], refl }
 
+lemma comp_range : (g.comp f).range = add_subgroup.map g.to_add_monoid_hom f.range :=
+begin
+  erw add_monoid_hom.map_range,
+  refl,
+end
+
+lemma mem_comp_range (x : V₁) : g (f x) ∈ (g.comp f).range :=
+begin
+  rw normed_group_hom.comp_range,
+  exact add_subgroup.mem_map_of_mem g.to_add_monoid_hom (set.mem_range_self x),
+end
+
+@[simp]
+lemma range_comp_incl_top : (f.comp (incl (⊤ : add_subgroup V₁))).range = f.range :=
```

Comment:
I think there should be a lemma `incl_range` that says that the range of `incl foo` is `foo`. Together with `comp_range` and `add_subgroup.map_top` (which hopefully exists) this should then be a one-line proof.